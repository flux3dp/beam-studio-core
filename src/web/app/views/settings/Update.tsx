import * as React from 'react';

import i18n from 'helpers/i18n';
import SelectControl from 'app/views/settings/SelectControl';
import { StorageKey } from 'interfaces/IStorage';

interface Props {
  updateNotificationOptions: { value: any, label: string, selected: boolean, }[];
  updateConfigChange: (id: StorageKey, newVal: any) => void;
}

function Update({ updateNotificationOptions, updateConfigChange }: Props): JSX.Element {
  const lang = i18n.lang;
  return (
    <>
      <div className="subtitle">{lang.settings.groups.update}</div>
      <SelectControl
        label={lang.settings.check_updates}
        options={updateNotificationOptions}
        onChange={(e) => updateConfigChange('auto_check_update', e.target.value)}
      />
    </>
  );
}

export default Update;
