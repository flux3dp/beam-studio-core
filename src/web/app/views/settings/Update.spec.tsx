import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

jest.mock('helpers/i18n', () => ({
  lang: {
    settings: {
      check_updates: 'Auto Check',
      groups: {
        update: 'Software Updates',
      },
    },
  },
}));

// eslint-disable-next-line import/first
import Update from './Update';

test('should render correctly', () => {
  const updateConfigChange = jest.fn();
  const wrapper = shallow(<Update
    updateNotificationOptions={[
      {
        value: 'TRUE',
        label: 'On',
        selected: true,
      },
      {
        value: 'FALSE',
        label: 'Off',
        selected: false,
      },
    ]}
    updateConfigChange={updateConfigChange}
  />);
  expect(toJson(wrapper)).toMatchSnapshot();

  wrapper.find('SelectControl').simulate('change', {
    target: {
      value: 'FALSE',
    },
  });
  expect(updateConfigChange).toHaveBeenCalledTimes(1);
  expect(updateConfigChange).toHaveBeenNthCalledWith(1, 'auto_check_update', 'FALSE');
});
