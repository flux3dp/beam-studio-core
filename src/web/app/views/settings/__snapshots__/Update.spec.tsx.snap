// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Fragment>
  <div
    className="subtitle"
  >
    Software Updates
  </div>
  <SelectControl
    label="Auto Check"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "On",
          "selected": true,
          "value": "TRUE",
        },
        Object {
          "label": "Off",
          "selected": false,
          "value": "FALSE",
        },
      ]
    }
  />
</Fragment>
`;
